app/controllers/mixins/containers_common_mixin.rb

```diff
@@ -210,6 +211,14 @@ def process_check_compliance(model, ids)
     end
   end
 
+  def check_smart_roles
+    %w(SmartProxy SmartState).each do |role|
+      unless MiqServer.all.any? { |s| s.has_active_role?(role.downcase) }
```

Comment:
@roliveri I am sorry for resurrecting this from the dead ;-)
I was trying to find out but couldn't actually find where we are making this decision. I think it might be in [1] but I am not sure because it seems too general of a function name.
I am asking because I want to answer to BZ [2] with a definitive answer.
I think what they need is to allow Jobs created in the UI zone that have no active roles to be moved to other zones, is that possible?

[1] https://github.com/ManageIQ/manageiq/blob/master/app/models/job_proxy_dispatcher.rb#L174
[2] https://bugzilla.redhat.com/show_bug.cgi?id=1459702